osquery/extensions/extensions.cpp

```diff
@@ -88,6 +100,56 @@ void ExtensionManagerWatcher::watch() {
   }
 }
 
+void loadExtensions() {
+  // Optionally autoload extensions
+  auto status = loadExtensions(FLAGS_extensions_autoload);
+  if (!status.ok()) {
+    LOG(WARNING) << "Could not autoload extensions: " << status.what();
+  }
+}
+
+void loadModules() {
+  auto status = loadModules(FLAGS_modules_autoload);
+  if (!status.ok()) {
+    LOG(WARNING) << "Modules autoload contains invalid paths";
+  }
+}
+
+Status loadExtensions(const std::string& paths) {
+  // Not implemented: Autoloading extensions given a search path.
+  return Status(0, "OK");
+}
+
+Status loadModulesFromDirectory(const std::string& dir) {
+  std::vector<std::string> modules;
+  if (!listFilesInDirectory(dir, modules).ok()) {
+    return Status(1, "Cannot read files from " + dir);
+  }
+
+  for (const auto& module_path : modules) {
+    // Silently allow module load failures to drop.
+    RegistryModuleLoader loader(module_path);
```

Comment:
This is a vulnerability, it must verify the permissions of the inode it's about to open. Permissions should be equal or greater to the effective permissions of the process. Otherwise, a lower-privileged user may add a library/dylib and inject/control the osquery process.